tock-process-console/src/board/connection.rs

```diff
@@ -0,0 +1,108 @@
+use anyhow::Error;
+use bytes::Bytes;
+use tokio::{
+    io::{split, AsyncReadExt, AsyncWriteExt, ReadHalf, WriteHalf},
+    sync::mpsc::{self, UnboundedReceiver, UnboundedSender},
+};
+use tokio_serial::{SerialPortBuilderExt, SerialStream};
+
+use super::{
+    decoder::{BoardMessage, Decoder},
+    event::{Event, NewMessageEvent},
+};
+
+pub struct ConnectionHandler {
+    port_reader: ReadHalf<SerialStream>,
+    port_writer: WriteHalf<SerialStream>,
+    event_writer: UnboundedSender<Event>,
+    command_reader: UnboundedReceiver<Bytes>,
+    decoder: Decoder,
+    decoded_receiver: UnboundedReceiver<BoardMessage>,
+}
+
+impl ConnectionHandler {
+    pub async fn new(
+        tty: &str,
+    ) -> Result<(UnboundedReceiver<Event>, UnboundedSender<Bytes>), Error> {
+        let mut port: SerialStream = tokio_serial::new(tty, 115200).open_native_async()?;
+        port.set_exclusive(false)?;
```

Comment:
This doesn't compile on Windows. add a `#cfg` directive to be excluded from windows

Reply:
i opted for a cfg! since its just one line